hs_core/views/utils.py

```diff
@@ -775,9 +775,56 @@ def move_or_rename_file_or_folder(user, res_id, src_path, tgt_path, validate_mov
 
     # ensure the target_full_path contains the file name to be moved or renamed to
     # if we are moving to a directory, put the filename into the request.
+    # TODO: #2105: this is not a sufficient check that the move is sane.
+    # TODO: #2105: separate the two use cases in this situation to avoid confusion
     if src_file_dir != tgt_file_dir and tgt_file_name != src_file_name:
         tgt_full_path = os.path.join(tgt_full_path, src_file_name)
 
+    # TODO: why is this optional?
+    if validate_move_rename:
+        # this must raise ValidationError if move/rename is not allowed by specific resource type
+        if not resource.supports_rename_path(src_full_path, tgt_full_path):
+            raise ValidationError("File/folder move/rename is not allowed.")
+
+    istorage.moveFile(src_full_path, tgt_full_path)
```

Comment:
The same ```imv``` iRODS command, i.e., ```istorage.moveFile()``` is used for both move and rename, so I doubt why we need to separate these two cases. It is advised that don't do move and rename at the same time by```imv```, i.e., not to do ```imv file1  coll1/file``` but do ```imv file1 coll1/file1``` to move the file, and do ```imv file1 file2``` to rename the file. I think it'd be good for us to reuse the same ```moveFile``` function for both move and rename like the current ```move_or_rename_file_or_folder()``` does although it may be improved upon as needed.